api/v1/databaseengine_types.go

```diff
@@ -73,6 +75,23 @@ type DatabaseEngineList struct {
 	Items           []DatabaseEngine `json:"items"`
 }
 
+// Versions struct represents available versions of database engine components.
+type Versions struct {
+	Engine map[string]*Component            `json:"engine,omitempty"`
+	Backup map[string]*Component            `json:"backup,omitempty"`
+	Proxy  map[string]map[string]*Component `json:"proxy,omitempty"`
+	Tools  map[string]map[string]*Component `json:"tools,omitempty"`
```

Comment:
If I understand the reasoning correctly, in the `Proxy` and `Tools` we have a `map[string]map[string]*Component` to first have the proxy/tool name and then the corresponding versions, right? Taking PXC as an example, we need this to accommodate both haproxy and proxysql and be able to show both, correct?

If my analysis is correct, should we also do the same for backup? Are you aware of any plans to support more than one backup tool? Maybe in PS operator?

Reply:
I know, that right now we don't really need it since PXC, PSMDB and PG only support one backup tool but wanted to throw that out there.